Address initial comments on updates to firmware

bparks13 opened the pull request:
- Find device address for methods based on the given port number
- Reorder parameters to be consistent with oni_* calls
- Before reading device IDs from an EEPROM, confirm that the first four bytes contain the characters "OESH"
- Make readBytes() more flexible, and create readEepromBytes() specifically for EEPROM byte reads
- Corrected logic for devices with a BNO but no EEPROM

Fixes #18

bparks13 committed:
Find device address for methods based on the given port number
- Reorder parameters to be consistent with oni_* calls
- Before reading device IDs from an EEPROM, confirm that the first four bytes contain the characters "OESH"
- Make readBytes() more flexible, and create readEepromBytes() specifically for EEPROM byte reads
- Corrected logic for devices with a BNO but no EEPROM

aacuevas commented on Source/devices/oni/AcqBoardONI.cpp:
```diff
@@ -738,28 +738,31 @@ void AcqBoardONI::scanPortsInThread()
+              {
+                evalBoard->setBnoAxisMap (i, 0b00100100);
+              }
+              else if (hasI2c[i])
```
Since we are using this check to set BNO axis maps, I'd use hasBNO instead of hasI2C here.

aacuevas commented on Source/devices/oni/rhythm-api/rhd2000ONIboard.cpp:
```diff
@@ -1204,43 +1206,80 @@ bool Rhd2000ONIBoard::isI2cCapable(oni_dev_idx_t device)
 }
 
-int Rhd2000ONIBoard::readByte(uint32_t address, oni_reg_addr_t i2cRawAddress, oni_reg_val_t* value, bool sixteenBitAddress)
+int Rhd2000ONIBoard::readByte (oni_dev_idx_t device, uint32_t address, oni_reg_val_t* value)
```
This is not what I meant. This is basically just a wrapper over oni_read_reg at this point with nothing added to it.
What I meant is `int Rhd2000ONIBoard::readI2CByte(oni_dev_idx_t device, uint32_t i2cDevAddress, oni_reg_addr_t i2cRegAddress, oni_reg_val_t* value, bool sixteenBitAddress)`
Just not have the eeprom harcoded here as it was before, but make this a general function to read bytes over i2c

then on `readEepromByte` you call this method with i2cDevAddress = 0x50 and sixteenBitAddress = true

aacuevas commented on Source/devices/oni/rhythm-api/rhd2000ONIboard.cpp:
```diff
@@ -1204,43 +1206,80 @@ bool Rhd2000ONIBoard::isI2cCapable(oni_dev_idx_t device)
+    oni_dev_idx_t i2cRawAddress = DEVICE_I2C_RAW_A + port * 2;
+
+    // NB: Confirm EEPROM first four bytes contains OESH
+    const char identifier[] = "OESH";
```
just to be sure, make it "OESH\x01" to include the spec version, in case we change it later, so even an outdated plugin would not break.

aacuevas reviewed (dismissed):
I made a couple of suggestions. Nothing code-breaking, so it could be merged as is, but look into addressing those if possible.

aacuevas commented on Source/devices/oni/rhythm-api/rhd2000ONIboard.cpp:
```diff
@@ -1204,43 +1206,80 @@ bool Rhd2000ONIBoard::isI2cCapable(oni_dev_idx_t device)
         return false;
 
     oni_reg_val_t val = 2; // Value == 2 means there is a BNO scan in progress
+    oni_dev_idx_t device = DEVICE_I2C_RAW_A + port * 2;
```
This is using the wrong address. It should use the bno address as base, since you are talking with the bno here, not the i2c raw.

bparks13 committed:
Addressed review comments
- Modified Rhd2000ONIBoard::readByte to be more generalized and flexible
- Expanded the EEPROM identifier to include "\x01", which is the spec version
- Corrected a bug that was accessing the wrong device index when checking if a BNO is connected

bparks13 committed:
Return patch version when getting firmware version
- Check what version is returned, and confirm if it is I2C capable

bparks13 commented:
@aacuevas Can you please review the changes made? I have addresses the review comments, as well as made the updates to the firmware version method described in #18, but I am unsure if I am checking the correct version number to see if the device is I2C capable.

aacuevas commented on Source/devices/oni/AcqBoardONI.cpp:
```diff
@@ -115,12 +115,10 @@ bool AcqBoardONI::detectBoard()
-            //    hasBNO[0] = true;
-            hasI2cSupport = major >= 2;
+            LOGC ("Open Ephys ECP5-ONI FPGA open. Gateware version v", major, ".", minor + "." + patch);
+            hasI2cSupport = major >= 1 && minor > 5;
```
This like this would fail with version 2.1 for example.
Although, thinking about it, I believe we should change it so this plugin only works with versions 0.x and 1.x
Version 2 is planned and will break API, So maybe if major > 2 it should report an error (and say to update the plugin) and return the same as board not detected., and then for 0.1 to 1.4 does not have i2c support, from 1.5 to 1.infinity, it does ;)
